pkg/cmd/gpg-key/add/http.go

```diff
@@ -61,6 +68,17 @@ func gpgKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader) e
 }
 
 func isDuplicateError(err *api.HTTPError) bool {
+	return err.StatusCode == 422 && len(err.Errors) == 2 &&
+		err.Errors[0].Field == "key_id" && err.Errors[0].Message == "key_id already exists"
+}
+
+func isKeyInvalidError(err *api.HTTPError) bool {
 	return err.StatusCode == 422 && len(err.Errors) == 1 &&
-		err.Errors[0].Field == "key" && err.Errors[0].Message == "key is already in use"
+		err.Errors[0].Field == "" && err.Errors[0].Message == "We got an error doing that."
+}
+
+func isGpgKeyArmored(keyBytes []byte) bool {
+	buf := make([]byte, 36)
+	copy(buf, keyBytes)
+	return bytes.Equal(buf, []byte("-----BEGIN PGP PUBLIC KEY BLOCK-----"))
```

Comment:
I am not an expert on armored keys or anything but does it also make sense to check that the footer `-----END PGP PUBLIC KEY BLOCK-----` is present?